src/test/java/io/jenkins/plugins/analysis/warnings/RecorderITest.java

```diff
@@ -0,0 +1,227 @@
+
+    /**
+     * Verify the info page of the Java parser with no warnings as input.
+     * Verify the health report (healthy=1, unhealthy=9) configured by the web UI has a score of 100
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void noWarningsWithHealthReport100() throws IOException {
+        verifyInfoPage("javac_no_warning.txt", 0, 100, true);
+    }
+
+    /**
+     * Verify the info page of the Java parser with no warnings as input and no configured health report.
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void noWarningsWithoutHealthReport() throws IOException {
+        verifyInfoPage("javac_no_warning.txt", 0, 0, false);
+    }
+
+    /**
+     * Verify the info page of the Java parser with one warning as input.
+     * Verify the health report (healthy=1, unhealthy=9) configured by the web UI has a score of 90
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void oneWarningWithHealthReport90() throws IOException {
+        verifyInfoPage("javac_one_warning.txt", 1, 90, true);
+    }
+
+    /**
+     * Verify the info page of the Java parser with one warning as input and no configured health report.
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void oneWarningWithoutHealthReport() throws IOException {
+        verifyInfoPage("javac_one_warning.txt", 1, 0, false);
+    }
+
+    /**
+     * Verify the info page of the Java parser with nine warnings as input.
+     * Verify the health report (healthy=1, unhealthy=9) configured by the web UI has a score of 10
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void nineWarningsWithHealthReport10() throws IOException {
+        verifyInfoPage("javac_9_warnings.txt", 9, 10, true);
+    }
+
+    /**
+     * Verify the info page of the Java parser with nine warnings as input and no configured health report.
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void nineWarningsWithoutHealthReport() throws IOException {
+        verifyInfoPage("javac_9_warnings.txt", 9, 0, false);
+    }
+
+    /**
+     * Verify the info page of the Java parser with ten warnings as input.
+     * Verify the health report (healthy=1, unhealthy=9) configured by the web UI has a score of 0
+     * @throws IOException when something went wrong with the web ui access
+     */
+    @Test
+    public void tenWarningsWithHealthReport0() throws IOException {
+        verifyInfoPage("javac_10_warnings.txt", 10, 0, true);
+    }
+
+    /**
+     * Verify the info page of the Java parser with ten warnings as input and no configured health report.
+     * @throws IOException when something went wrong with the web ui access
```

Comment:
CheckedExceptions are evil ;-) Am besten die Exception in der Methode fangen und wrappen, dann muss man nicht den Kommentar an allen Tests wiederholen.